csharp/summary.go

```diff
@@ -0,0 +1,46 @@
+package csharp
+
+import "fmt"
+
+type SummaryDeclaration struct {
+	description string
+	params      []map[string]string
```

Comment:
Had to use this as array of map because [Go maps](https://go.dev/blog/maps) doesn't iterate on the same order each execution time, so our generated code could have the summary params different each time we build.

From Go docs:
```
When iterating over a map with a range loop, the iteration order is not specified and is not guaranteed to be the same from one iteration to the next. 
````

Reply:
Nit (feel free to ignore): you could consider an slice of struct here. They could be package private, something like:

```go
type Param struct {
	name, description string
}

type SummaryDeclaration struct {
	params []Param
}
```